benchmarks/src/main/scala/zio/internal/MailboxBenchmark.scala

```diff
@@ -0,0 +1,131 @@
+package zio.internal
+
+import org.openjdk.jmh.annotations._
+
+import java.util.concurrent.TimeUnit
+
+/**
+ * Defines (asymetric) benchmarks to measure the relative performance of MPSC
+ * queues in the presence of concurrent operations. The benchmarks are organized
+ * into groups where each group consists of one or more producer threads that
+ * enqueue an element and a consumer thread that dequeues all elements.
+ *
+ * The total number of elements transiting the queue in an iteration is
+ * `number_of_producers * batch_size * operations_per_invocation`.
+ *
+ * In an ideal system, one that scales linearly with the number of threads, the
+ * measurement for all groups is expected to be the same.
+ *
+ * A `SingleShotTime` mode is used because
+ *   - a precise number of elements need to be dequeued
+ *   - the operation does not have a steady state
+ *   - the operation measurement is too small
+ */
+@BenchmarkMode(Array(Mode.SingleShotTime))
+@Fork(1)
+@OperationsPerInvocation(1)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@State(Scope.Group)
+@Measurement(batchSize = 1000000, iterations = 20)
```

Comment:
a bit unrealistic, we know the inbox is usually empty and actually try to make sure poll on empty mailbox is as efficient as possible

Reply:
I am assuming, in real usage, all enqueued elements are dequeued. The benchmark attempts to measure the performance of this use case.
What is the alternative here?

Reply:
Added `isEmpty` call in `poll`.